docs/design/2020-01-24-collations.md

```diff
@@ -0,0 +1,285 @@
+
+### Current collations in MySQL
+
+In MySQL, collation is treated as an attribute of the character set: for each character set has options for different collations(and one of them is the default collation), every collation belongs to one character set only. The effects of collation are:
+
+  * It defines a total order on all characters of a character set.
+  * It determines if padding is applied when comparing two strings.
+  * It describes comparisons of logically identical but physically different byte sequences.
+Some examples of the effects of collation can be found in [this part](https://dev.mysql.com/doc/refman/8.0/en/charset-collation-effect.html) of the MySQL Manual and [this webpage](http://demo.icu-project.org/icu-bin/locexp?_=en_US&x=col).
+
+Several collation implementations are described in [this part](https://dev.mysql.com/doc/refman/8.0/en/charset-collation-implementations.html) of MySQL Manual:
+
+  * Per-character table lookup for 8-bit character sets, for example: `latin1_swedish_ci`.
+  * Per-character table lookup for multibyte character sets, including some of the Unicode collations, for example: `utf8mb4_general_ci`.
+    - Full UCA implementation for some Unicode collations, for example: `utf8mb4_0900_ai_ci`.
+
+### Current collations in TiDB
+
+For all character sets that are supported, TiDB accepts all of their collations by syntax and stores them as a part of metadata. The real collate-related behaviors are always in binary. For example,
+
+```
+tidb> create table t(a varchar(20) charset utf8mb4 collate utf8mb4_general_ci key);
+Query OK, 0 rows affected
+tidb> show create table t;
++-------+-------------------------------------------------------------+
+| Table | Create Table                                                |
++-------+-------------------------------------------------------------+
+| t     | CREATE TABLE `t` (                                          |
+|       |   `a` varchar(20) COLLATE utf8mb4_general_ci NOT NULL,      |
+|       |   PRIMARY KEY (`a`)                                         |
+|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+-------------------------------------------------------------+
+1 row in set
+tidb> insert into t values ('A');
+Query OK, 1 row affected
+tidb> insert into t values ('a');
+Query OK, 1 row affected // Should report error "Duplicate entry 'a'"
+```
+In the case above, the user creates a column with a case-insensitive collation `utf8mb4_general_ci`. Since the 'real' collation of TiDB is always `utf8mb4_bin`, inserting data into the primary key with "A" and "a" does not lead to an error.
```

Comment:
Since the 'real' collation of TiDB is always `utf8mb4_bin`,
s/utf8mb4_bin/binary?

Reply:
Addressed.